src/node_contextify.cc

```diff
@@ -1564,9 +1564,13 @@ Local<Object> ContextifyContext::CompileFunctionAndCacheResult(
     return Object::New(env->isolate());
 
   std::unique_ptr<ScriptCompiler::CachedData> new_cached_data;
-  if (produce_cached_data) {
-    new_cached_data.reset(ScriptCompiler::CreateCodeCacheForFunction(fn));
+  if (produce_cached_data && !fn.IsEmpty()) {
+    TryCatchScope try_cache(env);
+    if (options != ScriptCompiler::kConsumeCodeCache) {
```

Comment:
I don't think this guard is the correct solution. When `produceCachedData` is true, a new cached data should always be returned regardless of whether the old one is consumed or not.